rs/sns/governance/src/governance.rs

```diff
@@ -2603,12 +2609,72 @@ impl Governance {
 
     async fn perform_execute_extension_operation(
         &self,
-        _execute_extension_operation: ExecuteExtensionOperation,
+        execute_extension_operation: ExecuteExtensionOperation,
     ) -> Result<(), GovernanceError> {
-        Err(GovernanceError::new_with_message(
-            ErrorType::InvalidCommand,
-            "ExecuteExtensionOperation is not supported yet.",
-        ))
+        let ValidatedExecuteExtensionOperation {
+            extension_canister_id,
+            operation_name,
+            operation_arg,
+        } = execute_extension_operation.try_into().map_err(|err| {
+            GovernanceError::new_with_message(
+                ErrorType::InvalidProposal,
+                format!("Invalid ExecuteExtensionOperation proposal: {:?}", err),
+            )
+        })?;
+
+        validate_execute_extension_operation(
+            &*self.env,
+            self.proto.root_canister_id_or_panic(),
+            extension_canister_id,
+            operation_name,
+            &operation_arg,
+        )
+        .await?;
+
+        let treasury_manager_canister_id = extension_canister_id;
+
+        let (arg, sns_amount_e8s, icp_amount_e8s) = self
+            .construct_treasury_manager_deposit_payload(operation_arg)
+            .await?;
+
+        self.deposit_treasury_manager(treasury_manager_canister_id, sns_amount_e8s, icp_amount_e8s)
+            .await?;
+
+        let balances = self
+            .env
+            .call_canister(treasury_manager_canister_id, "deposit", arg)
```

Comment:
Governance is calling directly, but with no timeouts.  This shouldn't be allowed.  Could we add a call_canister_with_timeout?